src/stream-tcp.c

```diff
@@ -927,6 +928,9 @@ static int StreamTcpPacketStateNone(ThreadVars *tv, Packet *p,
         if (!stream_config.midstream && stream_config.async_oneside == FALSE)
             return 0;
```

Comment:
If we want the midstream-policy to be non `midstream=true` dependant, then I'm thinking of adding a check for `midstream_policy == EXCEPTION_POLICY_IGNORE` here and with the other check. 

Is that acceptable?

Reply:
yeah, but mixing with `EXCEPTION_POLICY_PASS` should also be supported, as that still does stream reassembly and app-layer parsing/logging, just no detection (like `pass` rule keyword).